### PR TITLE
implemented anyfrax crosschain link if user has anyFrax on other chains

### PR DESCRIPTION
**Pull request by prodvdc**

### CODE REVIEW

**Comment by saito-sv** [comment]
This PR is automatically being deployed to https://preview-task-david-anyswap-frax.dev-app.axial.exchange

**Comment by ghost** [comment]
We don't need all this code for other chains, keep it Avalanche only, otherwise there will be a lot of unused code.

File: src/constants/index.ts [hunk 1]
```diff
@@ -655,3 +655,39 @@ export const ZERO_ADDRESS = "0x0000000000000000000000000000000000000000"
...
+  [EVM_CHAINIDS.MOONRIVER]: 'https://moonriver.api.onfinality.io/public',
+  [EVM_CHAINIDS.BOBA]: 'https://mainnet.boba.network',
+  [EVM_CHAINIDS.HARMONY]: 'https://api.harmony.one',
+}
+
+export const FRAX_ADDRESSES: { [key: string]: string }  = {
+  [EVM_CHAINIDS.ETHEREUM]: '0x853d955aCEf822Db058eb8505911ED77F175b99e',
+  [EVM_CHAINIDS.BSC]: '0x90C97F71E18723b0Cf0dfa30ee176Ab653E89F40',
+  [EVM_CHAINIDS.MATIC]: '0x45c32fA6DF82ead1e2EF74d17b76547EDdFaFF89',
+  [EVM_CHAINIDS.AVALANCHE]: '0xD24C2Ad096400B6FBcd2ad8B24E7acBc21A1da64',
```
Comment (ghost) [inline src/constants/index.ts:687]: The token address is wrong, AnyFRAX is 0xDC42728B0eA910349ed3c6e1c9Dc06b5FB591f98

File: src/pages/deposit/Deposit.tsx [hunk 2]
```diff
@@ -40,13 +40,163 @@ function Deposit({ poolName }: Props): ReactElement | null {
...
+      }
+    } catch (err) {
+      console.error('[fetchFraxBalanceOnHARMONY] ==> err', err)
+    }
+  }
+
+  
+  useEffect(() => {
+    const fetchFraxBalance = () => {
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
```
Comment (saito-sv) [inline src/pages/deposit/Deposit.tsx:176]: This is not what we want to do here, please find a better way to handle this call in a more logical manner.

File: src/pages/deposit/Deposit.tsx [hunk 3]
```diff
@@ -40,13 +40,163 @@ function Deposit({ poolName }: Props): ReactElement | null {
...
+      }
+    } catch (err) {
+      console.error('[fraxContractOnMOONRIVER] ==> err', err)
+    }
+  }
+
+  const fetchFraxBalanceOnBOBA = async () => {
+    try {
+      if (fraxContractOnBOBA && account) {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call
```
Comment (saito-sv) [inline src/pages/deposit/Deposit.tsx:148]: Lets not abuse the `disabling` of `lint` linter is here to help us write better code not to fight with it.

File: src/constants/index.ts (hunk 1 above)
Comment (saito-sv) [reply src/constants/index.ts:687]: @prodvdjin  if this is the wrong token, How are you even testing to make sure it works?

**Review by saito-sv — CHANGES_REQUESTED** [review]

**Comment by saito-sv** [comment]
Closing due to conflict and author no longer around.

**Pull request closed without merging**